vernac/mltop.mli

```diff
@@ -73,12 +73,16 @@ val module_is_known : string -> bool
 val add_known_module : string -> unit
 (* EJGA: Todo, this could take a PluginSpec.t at some point *)
 
-(** Declare a plugin plus a Coq-specific initialization function.
+(** Declare a initialization function.
     The initialization function is granted to be called after Coq is fully
-    bootstrapped, even if the plugin is statically linked with the toplevel *)
-val add_known_plugin : (unit -> unit) -> string -> unit
-
-(** Calls all initialization functions in a non-specified order *)
+    bootstrapped, even if the plugin is statically linked with the toplevel.
+    TODO consider the spec of this more carefully, pretty sure the above comment is incorrect or incomplete.
+*)
+val add_init_function : string -> (unit -> unit) -> unit
```

Comment:
We use this init function mechanism to ensure that the grammar modifications are re-applied if we backtrack and redo the Declare ML Module / Require.
We could use the declare_cache_obj stuff instead but that would be more complex and I don't think it would be more useful.

Reply:
At some point the init functions should take a pointer to the parsing state I understand, right?

Reply:
It's possible that we will do that but not in my short term plans